target/linux/ramips/image/mt7621.mk

```diff
@@ -235,6 +235,22 @@ define Device/k2p
 endef
 TARGET_DEVICES += k2p
 
+define Device/mir3p
+  DTS := MIR3P
+  BLOCKSIZE := 128k
+  PAGESIZE := 2048
+  KERNEL_SIZE:= 4096k
+  UBINIZE_OPTS := -E 5
+  IMAGE_SIZE := $(ralink_default_fw_size_32M)
+  DEVICE_TITLE := Xiaomi Mi Router 3 Pro
+  IMAGES += factory.bin
+  IMAGE/sysupgrade.bin := sysupgrade-tar | append-metadata
+  IMAGE/factory.bin := append-kernel | pad-to $$(KERNEL_SIZE) | append-ubi | check-size $$$$(IMAGE_SIZE)
+  DEVICE_PACKAGES := \
+  	kmod-usb3 kmod-usb-ledtrig-usbport wpad-basic uboot-envtools
+endef
+TARGET_DEVICES += mir3p
+
 define Device/mir3g
```

Comment:
@iscilyas It's additional (and optional) work, but since you're adding similar device, it would be nice if you could invest some time to cleanup and convert `mir3g` to the new DTS based board name detection as well (as you've done it for `mir3p`):

* convert `mir3g` ocurrencies to `xiaomi,mir3g`, removing it from `ramips_board_detect` etc.
* add `SUPPORTED_DEVICES += mir3g` for backward compatible sysupgrade

Thanks! :-)

Reply:
@ynezz i entirely agree it's a good idea to clean up mir3g as well (and i actually have one i can test). I think that would be better done in an additional (follow-up) patch. If it's OK (and since you already called it "optional" ;)) I'd prefer to limit this push to only mir3p support.

Reply:
Yes, please. This PR is about the mir3p. Let's keep it this way.